companion/embedding_service/main.py

```
@@ -0,0 +1,356 @@
+            texts_to_generate = [
+                text for text, cached in zip(chunk, cached_flags[-len(chunk):]) if not cached
+            ]
+            
+            if texts_to_generate:
+                logger.info(f"Generating embeddings for {len(texts_to_generate)} texts...")
+                
+                # Using thread pool for batch generation
+                def generate_batch_embeddings():
+                    return genai.embed_content(
+                        model="models/embedding-001",
+                        content=texts_to_generate,
```

Ruff (0.13.2)

Function definition does not bind loop variable `texts_to_generate`

(B023)